### PR TITLE
Support objects as input to test.each

### PR DESCRIPTION
**Pull request by ventuno**
Follow up on https://github.com/qunitjs/qunit/issues/1568#issuecomment-835627337 to add support for objects as input to `test.each`.

Screen Shot 2021-05-16 at 7 09 04 PM

@Krinkle, @smcclure15 let me know what your thoughts are on this change.

### CODE REVIEW

File: src/test.js [hunk 1]
```diff
@@ -746,10 +746,15 @@ function makeEachTestName( testName, argument ) {
...
+	} else if ( typeof data === "object" && data !== null ) {
+		const keys = Object.keys( data );
+		keys.forEach( ( key ) => {
+			eachFn( data[ key ], key );
+		} );
 	} else {
 		throw new Error(
-			`test.each expects an array of arrays or an array of primitives as the expected input.
-${typeof data} was found instead.`
+			`test.each expects an array of arrays, an array of primitives, or an object
```
Comment (Krinkle) [inline src/test.js:756]: I think the array/primitive distinction might be best omitted here now that we don't use variadic args. Something like "... expect an array or object as input."

**Review by Krinkle — CHANGES_REQUESTED** [review]
LGTM 👍. I've fiddled with the docs in b3521d052ba9d990d748a610a53645f621a4318f, so I'm afriad this needs a rebase (perhaps make the "object as data" case its own example?), but otherwise good to go!

File: src/test.js (hunk 1 above)
Comment (ventuno) [reply src/test.js:756]: After rebasing with your changes, rewrote this as "Array or object of data values passed to each test case" let me know if that works for you.

**Comment by ventuno** [comment]
@Krinkle, thanks do you mind taking another look?

File: src/test.js (hunk 1 above)
Comment (Krinkle) [reply src/test.js:756]: Sounds good to me! I don't see the change applied, however.

**Pull request merged**